snuba/datasets/tags_column_processor.py

```diff
@@ -169,41 +164,13 @@ def __extract_top_level_tag_conditions(self, condition: Expression) -> List[str]
         [[['tagskey' '=' 'a'], ['col2' '=' 'b']], ['tagskey' '=' 'c']] does not
         """
 
-        if is_binary_condition(condition, ConditionFunctions.EQ):
```

Comment:
All this depended on this expressions to be present in the AST:
```
arrayJoin(None, Column(None. None, "tags.key"))
```

Now when we get here we have already ran the processor introduced in #982 and wired up in this PR. This means that noe in the AST we would find those gigantic optimized expressions. So the easiest way to keep this working was to break the dependency on the AST and rely on the old representation, for the column_expr code.